number-46/index.html

```diff
@@ -0,0 +1,76 @@
+<!DOCTYPE html>
+<html>
+  <head>
+  </head>
+  <body>
+    <script>
+      class Shape {
+        constructor(length, width) {
+          this.length = length;
+          this.width = width;
+        }
+
+        get area() {
+          return this.length * this.width;
+        }
+
+        get name() {
+          return "Default Shape";
+        }
+      }
+
+      class Square extends Shape {
+        constructor(length) {
+          super(length, length);
+        }
+
+        get name() {
+          return "Square";
+        }
+      }
+
+      class Rectangle extends Shape {
+        constructor(length, width) {
+          super(length, width);
+        }
+
+        get name() {
+          return "Rectangle";
+        }
+      }
+
+      class Triangle extends Shape {
+        constructor(length, width) {
+          super(length, width / 2);
```

Comment:
🤔 , is it necessary to divide the width by 2? If I gave you this triangle and told you the width was 10cm and the height was 6cm. Would you expect the width to be 5?

<img src="https://technology240blog.files.wordpress.com/2016/03/length-and-area5.gif?w=600&h=379"/>